controllers/client/grafana_client.go

```diff
@@ -167,3 +169,49 @@ func NewGrafanaClient(ctx context.Context, c client.Client, grafana *v1beta1.Gra
 
 	return grafanaClient, nil
 }
+
+func NewGeneratedGrafanaClient(ctx context.Context, c client.Client, grafana *v1beta1.Grafana) (*genapi.GrafanaHTTPAPI, error) {
+	var timeout time.Duration
+	if grafana.Spec.Client != nil && grafana.Spec.Client.TimeoutSeconds != nil {
+		timeout = time.Duration(*grafana.Spec.Client.TimeoutSeconds)
+		if timeout < 0 {
+			timeout = 0
+		}
+	} else {
+		timeout = 10
+	}
+
+	credentials, err := getAdminCredentials(ctx, c, grafana)
+	if err != nil {
+		return nil, err
+	}
+
+	gURL, err := url.Parse(grafana.Status.AdminUrl)
+	if err != nil {
+		return nil, fmt.Errorf("parsing url for client: %w", err)
+	}
+
+	cfg := &genapi.TransportConfig{
+		Schemes:  []string{gURL.Scheme},
+		BasePath: "/api",
+		Host:     gURL.Host,
+		// APIKey is an optional API key or service account token.
+		APIKey: credentials.apikey,
+		// NumRetries contains the optional number of attempted retries
+		NumRetries: 0,
+	}
+	if credentials.username != "" {
+		cfg.BasicAuth = url.UserPassword(credentials.username, credentials.password)
+	}
+	cl := genapi.NewHTTPClientWithConfig(nil, cfg)
+	transport := cl.Transport.(*httptransport.Runtime)
+	wrapped := transport.Transport
+	transport.Transport = &instrumentedRoundTripper{
+		wrapped:         wrapped,
+		metric:          metrics.GrafanaApiRequests,
+		relatedResource: grafana.Name,
+	}
+	cl.SetTransport(transport)
```

Comment:
This part is pretty ugly and not yet using the timeout as specified. I'm working on a PR for the upstream client to improve this